lib/sparql_to_sw_solr/instance_solr_doc.rb

```diff
@@ -35,10 +37,23 @@ def solr_doc_hash
         return if CKEY_BLACKLIST.include?(@ckey)
         doc = init_doc
         add_doc_title_fields(doc)
+        add_doc_topic_fields(doc)
         doc
       end
     end
 
+    def sparql_prefixes
+      @sparql_prefixes ||= begin
+        bf = "PREFIX bf: <#{BF_NS}>"
+        mads = "PREFIX madsrdf: <#{MADSRDF}>"
+        [bf, mads].join("\n")
+      end
+    end
+
+    def sparql_query(query)
+      sparql.query(sparql_prefixes + query)
+    end
+
```

Comment:
I'm not entirely convinced about this refactoring.  To me, it adds to the cognitive load to understand the code, on a few fronts:

- I would be jumping around in the code to figure out where the namespace prefixes are included in the sparql, especially since the prefixes in the query are defined as a method in one file, and the rest of the sparql query can be in another method in a different file.

- the sparql_query  method, if we keep it, is poorly named, as it takes a query but returns an RDF::Solutions object.   "primary_title_result" in the instance_title_fields captured the idea that the method returned a result.  With these changes, `primary_title_result` returns a string that is part of a sparql query, and the actual query, which gets an RDF::Solutions object, is in a method named `sparql_query`.   To me, this is really confusing.

Also, the mads prefix will not be needed for all sparql queries  (We didn't need it for title or language, for example;  it's possible it's used in more than subject data -- I dunno.)

For now, I would stick the MADS_NS declaration in the `subject_fields.rb` file, and I would stick to the pattern we used for title_fields.   Will we need to refactor eventually?  Probably.  But let's wait and make sure we really know what makes sense in the broader context before we do it.